sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py

```diff
@@ -111,6 +113,8 @@ class ServiceBusReceiver(collections.abc.AsyncIterator, BaseHandler, ReceiverMix
     :paramtype receive_mode: Union[~azure.servicebus.ServiceBusReceiveMode, str]
     :keyword Optional[float] max_wait_time: The timeout in seconds between received messages after which the receiver
      will automatically stop receiving. The default value is None, meaning no timeout.
+     If there is a sessionful queue or topic and NEXT_AVAILABLE_SESSION is specified,
+     max_wait_time will replace the server-side operation timeout when waiting on connecting to a session.
```

Comment:
Would setting max_wait_time result in the same behavior (exiting the receive iterator fully) whether using sessions or not? If so, I think using max_wait_time as the operation_timeout makes sense. Otherwise, I think we would need to add a separate keyword.

One potential point of confusion in the docs is that we're saying "max_wait_time will replace the ... timeout". However, that value is None by default, while the operation timeout by default is 65 seconds. So I think we'd need to clarify that it would only replace the server-side op timeout if a float value is passed in.

Reply:
From local testing iterator has the same behavior with and without setting com.microsoft:timeout (i.e using NEXT_AVAILABLE_SESSION and not).

Reply:
I think we can use the flow I posted in the chat to add some clarity, basically `max_wait_time` on non-session based queues & subs waits for that time between messages received ( single or batch ). On session based queues that value is sent over as a link property to the service and now it handles that timeout for us. 

So for `receive_messages` & iter for subs, queues ( session based ) its max_wait_time that dictates how long to wait for the first message or between messages.